IFTTT SDK/SignInWithAppleAuthentication.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  SignInWithAppleAuthentication.swift
+//  IFTTTConnectSDK
+//
+//  Copyright © 2021 IFTTT. All rights reserved.
+//
+
+import AuthenticationServices
+
+/// Wraps a `ASAuthorizationController`. Used for requests to Sign In With Apple.
+@available(iOS 13.0, *)
+final class AppleSignInWebService: ServiceAuthentication {
+    
+    /// Acts as the primary handler to deal with authorization completion and errors from an `ASAuthorizationController`.
+    private class AuthenticationSessionAuthorizationHandler: NSObject, ASAuthorizationControllerDelegate {
+        /// The completion to be called upon success or error of the flow.
+        private let completion: ((Result<Completion, ErrorType>) -> Void)
+        
+        /// Creates an instance of `AuthenticationSessionAuthorizationHandler`.
+        ///
+        /// - Parameters:
+        ///     - completion: A closure to execute upon success or error of the flow.
+        init(completion: @escaping ((Result<Completion, ErrorType>) -> Void)) {
+            self.completion = completion
+            super.init()
+        }
+        
+        @available(iOS 13.0, *)
+        func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+            switch authorization.credential {
+            case let appleIDCredential as ASAuthorizationAppleIDCredential:
+                guard let identityTokenData = appleIDCredential.identityToken,
+                    let identitityCodeString = String(data: identityTokenData, encoding: .utf8) else {
+                    completion(.failure(.invalidResponse))
+                    return
+                }
+                completion(.success(identitityCodeString))
+            default:
+                completion(.failure(.invalidResponse))
+            }
```

Comment:
think this can be collapsed into a single `guard`,

```swift
guard
    let appleIDCredential = authorization.credential as? ASAuthorizationAppleIDCredential,
    let identityTokenData = appleIDCredential.identityToken,
    let identitityCodeString = String(data: identityTokenData, encoding: .utf8) else {
        completion(.failure(.invalidResponse))
        return
}

completion(.success(identitityCodeString))
```